projects/kit/components/segmented/segmented.directive.ts

```diff
@@ -42,12 +43,13 @@ export class TuiSegmentedDirective implements AfterContentChecked, AfterContentI
     public ngAfterContentInit(): void {
         tuiQueryListChanges(this.controls)
             .pipe(
-                switchMap((controls) => merge(controls.map((c) => c.valueChanges))),
+                switchMap((controls) => merge(controls.map((c) => tuiControlValue(c)))),
+                mergeAll(),
                 filter(() => this.isBrowser),
                 takeUntilDestroyed(this.destroyRef),
             )
-            .subscribe(() => {
-                this.update(this.el.querySelector(':checked'));
+            .subscribe((value) => {
+                this.update(this.el.querySelector(`[value=${value}]`));
```

Comment:
Why this change?

Reply:
Yeh, you are right, we don't need this change.

Reply:
And then `mergeAll()` is unnecessary too

Reply:
we still need mergeAll here, tested locally.